pilot/pkg/networking/core/tracing.go

```diff
@@ -717,3 +720,15 @@ func buildHTTPHeaders(headers []*meshconfig.MeshConfig_ExtensionProvider_HttpHea
 	}
 	return target
 }
+
+func buildInitialMetadata(metadata []*meshconfig.MeshConfig_ExtensionProvider_HttpHeader) []*core.HeaderValue {
+	target := make([]*core.HeaderValue, 0, len(metadata))
```

Comment:
This is always setting `initialmetadata` to non nil, can we add `if metadata == nil { return nil }`?

Reply:
Done. :)